webgoat-integration-tests/src/test/java/org/owasp/webgoat/IntegrationTest.java

```diff
@@ -0,0 +1,200 @@
+package org.owasp.webgoat;
+
+import io.restassured.RestAssured;
+import io.restassured.config.RestAssuredConfig;
+import io.restassured.config.SSLConfig;
+import lombok.Getter;
+import org.hamcrest.CoreMatchers;
+import org.junit.After;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.BeforeClass;
+import org.owasp.webwolf.WebWolf;
+import org.springframework.boot.builder.SpringApplicationBuilder;
+
+import java.io.IOException;
+import java.net.ServerSocket;
+import java.util.Map;
+import java.util.UUID;
+
+public abstract class IntegrationTest {
+
+    private static String WEBGOAT_URL = "http://localhost:8080/WebGoat/";
+    private static String WEBWOLF_URL = "http://localhost:9090/";
+
+    //This also allows to test the application with HTTPS when outside testing option is used
+    protected static RestAssuredConfig restConfig = RestAssuredConfig.newConfig().sslConfig(new SSLConfig().relaxedHTTPSValidation());
+
+    @Getter
+    private String webGoatCookie;
+    @Getter
+    private String webWolfCookie;
+    @Getter
+    private String webgoatUser = UUID.randomUUID().toString();
```

Comment:
Compared to the initial version, the user is really unique and new, but manual verification would be a bit harder. You can no longer override it with a user from a command line parameter. Also the database will keep on growing. Perhaps it is possible to clean the database for integration tests or use a different tmp folder

Reply:
Manual verification would still be desired in order to check for mistakes in the html. Which is sometime the cause of the exercise buttons to change color. Also because of tests are now split into more tests. Each test result is linked to 1 unique user. With an override user, you could see all the results for that one user.

Reply:
Yes I see the point, so the case you describe is, you run the test you login and see the UI actually updated? (Manually). I will add the option to override the username

Reply:
Yes that is what I meant. That would be great

Reply:
fixed, I opted for a default user which gives the benefits you mentioned above

Reply:
Can you explain how this works? Did you forget to commit this? I expected something like:

 = System.getProperty("WEBGOAT_USER", UUID.randomUUID().toString());

Reply:
@zubcevic the user is just fixed now, for the current test cases we have this works. Maybe in the future we need specific checks